### ISSUE TITLE
Option for solo-isolate

### ISSUE THREAD

**Issue by SameExpert** [LMMS/lmms#5834]
### Enhancement Summary
Sometimes I may want a channel or a track to be kept on, i.e. I don't want _solo_ to affect that particular channel or track. For example, if I want to listen only a particular synth track but along with drums and bass tracks, If I have an option to _solo-isolate_ them, I wouldn't need to switch on these tracks manually as I have enabled "solo-isolate" option for them.

If this option is available, _solo_ would not affect _solo-isolate_-enabled track or channel.

#### Justification
It saves time for a user when desired track or channel is kept on while soloing any track or channel.

#### Mockup

1. If I right-click on a _channel_ in the **FX Mixer** window, there would be a checkbox "**Solo-isolate**".
2. If I click on ![Screenshot_20201208_124433](https://user-images.githubusercontent.com/59734562/101452057-2887f680-3953-11eb-9897-53b6c23b9e09.png) button of a track in **Song Editor** or **Beat+Bassline Editor** window, there would be a checkbox "**Solo-isolate**".

**Comment by the-bebop** [comment]
I can take over this one.
For clarification, because I'm not sure about the difference of _solo_ and _solo-isolate_. From my understatement if an _element_ was marked as solo, this solo-_container_ should be extendable, it's like the vice-virsa of mute.

**Comment by qnebra** [comment]
It is possible to do in Song Editor, User can have several tracks unmuted when one is in 'solo' state. Same thing is in Mixer.

**Comment by the-bebop** [comment]
Correct, but currently it's only possible to have 1 solo element - which should be expanded with this contribution

**Comment by SameExpert** [comment]
> For clarification, because I'm not sure about the difference of _solo_ and _solo-isolate_. From my understatement if an _element_ was marked as solo, this solo-_container_ should be extendable, it's like the vice-virsa of mute.

_Soloing_ a track or tracks turns off all the other tracks, except _solo-isolated_ track(s).

> It is possible to do in Song Editor, User can have several tracks unmuted when one is in 'solo' state. Same thing is in Mixer.

Yes, but that I should do it every time I solo a track, manually. Whereas, if I _solo-isolate_ **Track 1**, every time I _solo_ **Track 2**, **Track 3** or **Track 4**, **Track 1** should be kept on only, if it was on. Soloing ANY other track should not affect **Track 1**, if it's on, it should remain on, even if I solo any other track.

**Comment by the-bebop** [comment]
I see. Shouldn't it then be better to create sort of persistent solo-groups (e.g Backgrounds, Drum- and Bassline, Synth- and Pad, ...)?

**Comment by Spekular** [comment]
> I see. Shouldn't it then be more convenient to create sort of persistent solo-groups (e.g Backgrounds, Drum- and Bassline, Synth- and Pad, ...)?

Which you can accomplish with the Mixer. I think the best way to resolve this issue is to ensure that soloed mixer channels are audible regardless of routing (right now if you send them to another channel that channel needs to be unmuted).

**Comment by SameExpert** [comment]
> I see. Shouldn't it then be more convenient to create sort of persistent solo-groups (e.g Backgrounds, Drum- and Bassline, Synth- and Pad, ...)?

I have a seperate channel for **Reverb**. Any channel (**Drums**, **Synth**, etc.) I connect to the **Reverb** channel should sound along with the reverb. If I solo any channel, the **Reverb** channel also gets muted. Every time I solo a track, I need to manually unmute the **Reverb** channel to get the reverb of the soloed track.

To understand solo-isolate better, please check these videos:
https://youtu.be/3L6UV7eA7M0
https://youtu.be/TBrPcEIVXks.

**Comment by qnebra** [comment]
lmms doesn't have separate Reverb channel like Ableton does, so this thing doesn't work here

**Comment by SameExpert** [comment]
> lmms doesn't have separate Reverb channel like Ableton does, so this thing doesn't work here

I created it. I added **Calf Reverb LADSPA** to an empty channel and named it "Reverb". In **Calf Reverb LADSPA**, I set **Wet Amount** to full. Then I sent the channels to the Reverb channel with desired amounts.

**Comment by coderman64** [comment]
Not sure if I should weigh in on this issue, but to me, it seems that "solo-isolate" seems to be a confusing term to use for this particular feature. While I haven't used many other DAWs personally, the referenced videos from ProTools seem to use "Solo-Safe" so it doesn't seem like this is a standardized name. 

Personally, I think that it should be called/labeled as "Ignore Solo" which might make it more clear to new users what it does. 

As for its usefulness, while not _strictly_ necessary (since individual tracks can be unmuted in solo) it seems like a handy feature that would save a user a few clicks and a bit of annoyance.

**Comment by SameExpert** [comment]
> it seems that "solo-isolate" seems to be a confusing term to use for this particular feature. While I haven't used many other DAWs personally, the referenced videos from ProTools seem to use "Solo-Safe" so it doesn't seem like this is a standardized name.

The names might be different but the thing that they do is the same. **FL Studio** has [_Mute / Solo lock state_](https://www.image-line.com/fl-studio-learning/fl-studio-online-manual/html/playlist.htm#:~:text=Lock%20state) [[video](https://youtu.be/pZtGQcFFLI0)],  **Waveform** has _Solo Isolate_ [[video](https://youtu.be/X-l92OOYgiM?t=271)], **Ardour** has both [Solo Isolate](https://manual.ardour.org/working-with-tracks/audio-track-controls#:~:text=Solo%20isolate) and [Solo Safe](https://manual.ardour.org/working-with-tracks/audio-track-controls#:~:text=Solo%20safe) (I don't know why) and **Pro Tools** has _Solo Safe_ [[video](https://youtu.be/3L6UV7eA7M0)]. So basically these two terms are well known.



> Personally, I think that it should be called/labeled as "Ignore Solo" which might make it more clear to new users what it does.

Yes, _Ignore Solo_ would be very precise (and also easier to understand).


> As for its usefulness, while not _strictly_ necessary (since individual tracks can be unmuted in solo) it seems like a handy feature that would save a user a few clicks and a bit of annoyance.

Yes, this would save clicks and comes handy. Unmuting individual tracks everytime when soloing a track would be really annoying sometimes.

**Comment by SameExpert** [comment]
This feature request is EXACTLY as same as when you UNCHECK ```Mute automation tracks during solo```, but for tracks other than automation.
